src/universe/visualization/img_tools.py

```diff
@@ -14,3 +18,313 @@ def rotate(img: np.ndarray, angle: float) -> np.ndarray:
     rot_mat = cv.getRotationMatrix2D((cols // 2, rows // 2), angle, 1)
 
     return cv.warpAffine(img, rot_mat, (cols, rows))
+
+class _Layer:
+    """
+    A layer that can be drawn on an image.
+    """
+
+    def __init__(self, color: Tuple[int, int, int], opacity: int, name: str = None):
+        """
+        Creates a new layer with the given color, opacity and name.
+
+        :param color: color of the layer (RGB)
+        :param opacity: opacity of the layer (0-100)
+        :param name: name of the layer
+        """
+        self._opacity = None
+        self.set_opacity(opacity)
+        self.color = color
+        self.name = name
+
+    def set_opacity(self, opacity: int):
+        """
+        Sets the opacity of the layer.
+
+        :param opacity: opacity of the layer (0-100)
+        """
+        assert 0 <= opacity <= 100, 'Alpha must be between 0 and 100'
```

Comment:
Asserts shouldn't be used as a form of expected error handling. Use assert only for conditions that should never occur (programming mistake, incorrect assumption). Asserts are mainly for development and debugging (also used in unit tests). In production (-O flag) asserts are removed from the code.

Reply:
If you want to enforce that the opacity variable is between 0 and zero, you might use raise an error, for example:
```suggestion
        if opacity < 0 or opacity > 100:
            raise ValueError('Opacity must be between 0 and 100')
```
Or you can use clipping silently:
```suggestion
        opacity = min(max(0, opacity), 100)
```

Reply:
I think in this case clipping AND a warning message would be optimal

Reply:
Clipping and printing a warning then